Test runs failing to start

When running tests in GitHub, the error below is happening.
The tests pass locally.
Need to figure this out.

Tests disabled in GH for now.

```
Run python -m pytest .
  python -m pytest .
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.10.14/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.14/x64/lib
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/_pytest/main.py", line [2](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:2)79, in wrap_session
INTERNALERROR>     config._do_configure()
INTERNALERROR>   File "/opt/hostedtoolcache/Python/[3](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:3).10.14/x64/lib/python3.10/site-packages/_pytest/config/__init__.py", line 1118, in _do_configure
INTERNALERROR>     self.hook.pytest_configure.call_historic(kwargs=dict(config=self))
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.1[4](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:4)/x64/lib/python3.10/site-packages/pluggy/_hooks.py", line [5](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:5)35, in call_historic
INTERNALERROR>     res = self._hookexec(self.name, self._hookimpls.copy(), kwargs, False)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.14/x[6](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:6)4/lib/python3.10/site-packages/pluggy/_manager.py", line 120, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 13[9](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:10), in _multicall
INTERNALERROR>     raise exception.with_traceback(exception.__traceback__)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.[10](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:11).14/x64/lib/python3.10/site-packages/pluggy/_callers.py", line 103, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pytestqt/plugin.py", line 243, in pytest_configure
INTERNALERROR>     qt_api.set_qt_api(config.getini("qt_api"))
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.14/x64/lib/python3.10/site-packages/pytestqt/qt_compat.py", line 109, in set_qt_api
INTERNALERROR>     self.QtGui = _import_module("QtGui")
INTERNALERROR>   File "/opt/hostedtoolcache/Python/3.10.[14](https://github.com/joaonc/qt_playground/actions/runs/10081043088/job/27872207177?pr=30#step:5:15)/x64/lib/python3.10/site-packages/pytestqt/qt_compat.py", line 105, in _import_module
INTERNALERROR>     m = __import__(_root_module, globals(), locals(), [module_name], 0)
INTERNALERROR> ImportError: libEGL.so.1: cannot open shared object file: No such file or directory
Error: Process completed with exit code 3.
```

--- a/.github/workflows/linting.yml
+++ b/.github/workflows/linting.yml
@@ -8,9 +8,10 @@ jobs:
   linting:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v4
       - name: Set up Python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5
         with:
           python-version: '3.10'
       - name: Install requirements

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -1,26 +1,33 @@
 name: tests
 
 on:
-#  - pull_request
-#  Fix on issue #31 https://github.com/joaonc/qt_playground/issues/31
-  noevent:
+  - pull_request
 
 jobs:
   pytest:
-    runs-on: ubuntu-latest
     strategy:
       matrix:
-        # Min and max versions supported
-        python-version: ['3.10', '3.11']
+        include:
+          - os: windows-latest
+            python-version: '3.11'
+          - os: macos-latest
+            python-version: '3.10'
+    runs-on: ${{ matrix.os }}
     name: Python ${{ matrix.python-version }}
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
+      - name: Install libEGL
+        # Implementation of the EGL (Embedded-Systems Graphics Library) API.
+        # No need to install on Windows or Mac.
+        run: |
+          sudo apt-get update -yqq
+          sudo apt-get install -y libegl1-mesa
+        if: runner.os == 'Linux'
       - name: Set up Python
         uses: actions/setup-python@v5
         with:
-          python-version: ${{ matrix.python-version }}
-          architecture: x64
+          python-version: '${{ matrix.python-version }}'
       - name: Install requirements
         run: |
           python -m pip install -U pip

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -14,9 +14,10 @@ jobs:
         python-version: ['3.10', '3.11']
     name: Python ${{ matrix.python-version }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout code
+        uses: actions/checkout@v4
       - name: Set up Python
-        uses: actions/setup-python@v3
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
           architecture: x64

--- a/.github/workflows/update-gh-pages.yml
+++ b/.github/workflows/update-gh-pages.yml
@@ -6,6 +6,7 @@ on:
       - main
     paths:
       - 'docs/**'
+      - 'mkdocs.yml'
 
 permissions:
   contents: write

--- a/.github/workflows/update-gh-pages.yml
+++ b/.github/workflows/update-gh-pages.yml
@@ -14,8 +14,9 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-python@v4
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
         with:
           python-version: 3.x
       - name: Install requirements

--- a/docs/release.md
+++ b/docs/release.md
@@ -10,21 +10,23 @@ Authenticate with this command:
 gh auth login
 ```
 
-Sample output from login with the `HTTPS` protocol and via web browser.
-```
-gh auth login
-? What account do you want to log into? GitHub.com
-? What is your preferred protocol for Git operations on this host? HTTPS
-? Authenticate Git with your GitHub credentials? Yes
-? How would you like to authenticate GitHub CLI? Login with a web browser
+??? Note "Sample output"
 
-! First copy your one-time code: 9999-9999
-Press Enter to open github.com in your browser... 
-✓ Authentication complete.
-- gh config set -h github.com git_protocol https
-✓ Configured git protocol
-✓ Logged in as <GH username>
-```
+    Sample output from login with the `HTTPS` protocol and via web browser.
+    ```
+    gh auth login
+    ? What account do you want to log into? GitHub.com
+    ? What is your preferred protocol for Git operations on this host? HTTPS
+    ? Authenticate Git with your GitHub credentials? Yes
+    ? How would you like to authenticate GitHub CLI? Login with a web browser
+
+    ! First copy your one-time code: 9999-9999
+    Press Enter to open github.com in your browser... 
+    ✓ Authentication complete.
+    - gh config set -h github.com git_protocol https
+    ✓ Configured git protocol
+    ✓ Logged in as <GH username>
+    ```
 
 You can authenticate in other ways, see
 [docs](https://cli.github.com/manual/gh_auth_login) for more info.
@@ -41,7 +43,7 @@ You can authenticate in other ways, see
 
 !!! Note "Recommended command"
 
-    Create the file `release_notes.md` and _don't_ add it to the project
+    Create the file `release_notes.md` and _don't_ add it to the project.
 
     ```
     inv build.release --notes-file release_notes.md
@@ -50,13 +52,13 @@ You can authenticate in other ways, see
 See sections below for beta and draft releases.
 
 #### Beta
-TODO
+**TODO**
 
 #### Draft
-TODO
+**TODO**
 
 ### Upload artifacts
-TODO
+**TODO**
 
 ## More info
 [Managing releases in a repository](https://docs.github.com/en/repositories/releasing-projects-on-github/managing-releases-in-a-repository).

